scripts/vale-review.sh

```diff
@@ -0,0 +1,94 @@
+    echo "$finalJson" > vale_output_temp.json
+
+    # Following logic checks if the line number is a part of the git diff. If it's not part of the diff it will be discarded.
+    # We only want to check new/modified content, plus the GitHub API only accepts comments within the diff for the review comments endpoint.
+    #echo "Checking line number is within the diff..."
+
+    # Iterate through the JSON objects
+    jq -c '.[]' vale_output_temp.json | while IFS= read -r object; do
+        #echo "$object"
+        BODY=$(echo "$object" | jq -r '.body')
+        FILENAME=$(echo "$object" | jq -r '.path')
+        LINE_NUMBER=$(echo "$object" | jq -r '.line')
+        
+        # Check the unified diff for the alert and file
+        FULL_DIFF=$(git diff --unified=0 --stat --diff-filter=AM HEAD~1 HEAD "${FILENAME}" ':(exclude)_unused_topics/*')
+        
+        # Iterate through each line to find the line diff info and check if the alert is in the diff
+        while read -r line; do
+            # Check if the line contains the hunk beginning with @@
+            if [[ $line =~ @@ ]]; then
+
+                # Valid:
+                # @@ -35 +31 @@
+                # @@ -35 +31,5 @@
+
+                # Check if there is a comma in the number pairing before @@
+                if [[ $line =~ \+.*\,.*\ @@ ]]; then
+                    # There are comma separated numbers before closing @@. Grab the number before the comma as the diff_start_line, after the comma is the added_lines. 
+                    added_lines=$(echo "$line" | grep -oP '\d+\s+@@' | grep -oP '\d+')
+                    diff_start_line=$(echo "$line" | awk -F'+' '{print $2}' | awk -F',' '{print $1}')
+                else
+                    # There are no comma seperated numbers. Consider the number after the plus as diff_start_line with no added lines - this means there's a modification on a single line
+                    added_lines=0
+                    diff_start_line=$(echo "$line" | grep -oP '\+\d+\s+@@' | grep -oP '\d+')
+                fi
+
+                # If the last_number is 0, disregard the hunk and move to the next hunk as zero lines were modified (deletions only)
+                if [ "$diff_start_line" -eq 0 ]; then
+                    continue
+                fi
+
+                # Check if the LINE_NUMBER falls within the range (diff_start_line) to (diff_start_line + added_lines)
+                if (( LINE_NUMBER >= diff_start_line && LINE_NUMBER <= diff_start_line + added_lines )); then
+                    
+                    post_variable_metadata "$BODY" "$FILENAME" "$LINE_NUMBER"
+                    
+                    break  # Exit the loop since the alert is within the diff, move on to the next JSON object
+                fi
+            fi
+                
+        done <<< "$FULL_DIFF"
+     
+    done
+
+done
+
+rm vale_output_temp.json
```

Comment:
nit - this isn't required in build env and anyone who runs this locally will want to see output